tests/neural_compressor/test_optimization.py

```diff
@@ -365,40 +399,6 @@ def test_one_shot(self):
         def compute_metrics(p: EvalPrediction):
             return metric.compute(predictions=np.argmax(p.predictions, axis=1), references=p.label_ids)
 
-        def train_func(model):
-            trainer.model_wrapped = model
-            trainer.model = model
-            _ = trainer.train(agent)
-            return trainer.model
-
-        def eval_func(model):
-            trainer.model = model
-            metrics = trainer.evaluate()
-            return metrics["eval_accuracy"]
-
-        config_path = os.path.dirname(os.path.abspath(__file__))
-        q8_config = IncQuantizationConfig.from_pretrained(config_path)
-        q8_config.set_config("quantization.approach", IncQuantizationMode.AWARE_TRAINING.value)
-        q8_config.set_config("tuning.accuracy_criterion.relative", 0.2)
-        q8_config.set_config("model.framework", "pytorch_fx")
-        quantizer = IncQuantizer(q8_config, eval_func=eval_func, train_func=train_func)
-        distillation_config = IncDistillationConfig.from_pretrained(config_path)
-        distiller = IncDistiller(
-            teacher_model=teacher_model,
-            config=distillation_config,
-            eval_func=eval_func,
-            train_func=train_func,
-        )
-        optimizer = IncOptimizer(
-            model,
-            quantizer=quantizer,
-            distiller=distiller,
-            one_shot_optimization=True,
-            eval_func=eval_func,
-            train_func=train_func,
```

Comment:
Why move this section ?

Reply:
Because IncQuantizer need calibration_dataloader, so it need trainer to generate a dataloader. And PyTorch 1.13 and above version, fx prepare function need a parameter: example_input, so we need send dataloader to INC for static quantization and quantzation-aware training.

Reply:
Thanks for the explanation @PenghuiCheng ! I added a suggestion on #95